src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -444,12 +502,17 @@ void doLastBlock(byte[] in, int inOfs, int len, byte[] out, int outOfs,
      * @return the number of bytes placed into the <code>out</code> buffer
      */
     int encrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
-        ArrayUtil.blockSizeCheck(len, blockSize);
-
         checkDataLength(processed, len);
 
         processAAD();
-
+        int remainder = len % blockSize;
+        if ( remainder > 0) {
+            if (ibuffer == null) {
+                ibuffer = new ByteArrayOutputStream(len % blockSize);
+            }
+            len -= remainder;
+            ibuffer.write(in, len, remainder);
```

Comment:
Any chance that 'ibuffer' already contains earlier buffered bytes? If 'ibuffer' contains bytes. then these need to be processed before processing 'in' and you can't write the to-be-processed bytes into 'ibuffer' here.

Reply:
I does not.  CipherCore only sends block sized data, and the ByteBuffer code uses this method during final operations.

Reply:
Hmm, ok, this is getting obscure and the correctness is depending on the calling pattern. Can we add a check on line 514 to ensure that there is no bytes in ibuffer?

Reply:
This got added into v3